src/dune_stats/dune_stats.ml

```diff
@@ -104,34 +104,76 @@ module Fd_count = struct
     | Unknown
     | This of int
 
-  let try_to_use_lsof () =
+  let lsof () =
     (* note: we do not use the Process module here, because it would create a
        circular dependency *)
-    let temp = Temp.create File ~prefix:"dune" ~suffix:"lsof" in
-    let stdout =
-      Unix.openfile
-        (Path.to_absolute_filename temp)
-        [ O_WRONLY; O_CREAT; O_TRUNC; O_SHARE_DELETE ]
-        0o666
+    let lsof_r, lsof_w = Unix.pipe ~cloexec:true () in
+    let pid =
+      let prog = "/usr/sbin/lsof" in
```

Comment:
I realize that this was in the original code, but shouldn't we look in PATH here?

Reply:
Why not? I suppose it could exist elsewhere on some BSD for example.

Reply:
yes, I was thinking of nix too.